packages/bpk-component-card-list/src/BpkCardListRowRail/BpkCardListCarousel.tsx

```diff
@@ -69,10 +73,44 @@ const BpkCardListCarousel = (props: CardListCarouselProps) => {
   const openSetStateLockTimeoutRef = useRef<NodeJS.Timeout | null>(null);
 
   const observerVisibility = useIntersectionObserver(
-    { root, threshold: 0.5 },
+    { root, threshold: 0.1 },
     setVisibilityList,
   );
 
+  useScrollToCard(
+    currentIndex * initiallyShownCards,
+    root,
+    cardRefs,
+    stateScrollingLockRef,
+  );
+
+  // Similar to Virtual Scrolling to improve performance
+  const firstVisibleIndex = Math.max(0, visibilityList.indexOf(1));
+  const lastVisibleIndex = firstVisibleIndex + initiallyShownCards - 1;
+  const renderList = visibilityList.map((_, index) =>
+    index >= firstVisibleIndex - RENDER_BUFFER_SIZE &&
+    index <= lastVisibleIndex + RENDER_BUFFER_SIZE
+      ? 1
+      : 0,
+  );
+
+  const cardRefCallback = (index: number) => (el: HTMLDivElement | null) => {
+    cardRefs.current[index] = el;
+    observerVisibility(el, index);
+    setA11yTabIndex(el, index, visibilityList);
+    // record the first card's width and height when it becomes visible
```

Comment:
The width and height of all cards in the Cardlist are consistent, including the placeholders for unrendered cards.